modules/core/03-connection/keeper/verify.go

```diff
@@ -218,20 +219,49 @@ func (k Keeper) VerifyPacketCommitment(
 		return errorsmod.Wrapf(clienttypes.ErrClientNotActive, "client (%s) status is %s", clientID, status)
 	}
 
-	// get time and block delays
-	timeDelay := connection.GetDelayPeriod()
-	blockDelay := k.getBlockDelay(ctx, connection)
+	var multihopProof commitmenttypes.MsgMultihopProofs
+	if len(connectionHops) > 1 {
```

Comment:
Here I merged multi-hop and single hop logic. If the proof is single hop then we create a multi-hop proof out of it, which is a degenerate case which should be handled exactly the same as a single hop proof (multi-hop proof with no consensus/connection proofs)!